### PR TITLE
Enable usage stats collection for ray.init iff nightly wheels

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
For nightly wheels, we want to collect usage stats for local clusters started via `ray.init()` as well.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/usage/usage_lib.py [hunk 1]
```diff
@@ -356,18 +360,23 @@ def _generate_cluster_metadata():
...
+    prompt_print = cli_logger.print if cli else print
+
     usage_stats_enabledness = _usage_stats_enabledness()
     if usage_stats_enabledness is UsageStatsEnabledness.DISABLED_EXPLICITLY:
-        cli_logger.print(usage_constant.USAGE_STATS_DISABLED_MESSAGE)
+        prompt_print(usage_constant.USAGE_STATS_DISABLED_MESSAGE)
     elif usage_stats_enabledness is UsageStatsEnabledness.ENABLED_BY_DEFAULT:
-
-        if cli_logger.interactive:
+        if not cli:
```
Comment (ericl) [inline python/ray/_private/usage/usage_lib.py:375]: It seems this path is only hit for nightly wheels. Consider renaming the cli flag and also the message to `is_nightly` / `...ENABLED_BY_DEFAULT_FOR_NIGHTLY_MESSAGE` for clarity.

File: python/ray/_private/usage/usage_constants.py [hunk 2]
```diff
@@ -8,16 +8,22 @@
...
+    " `ray disable-usage-stats` before `ray.init()`. "
+    "See https://docs.ray.io/en/master/cluster/usage-stats.html for more details."
+)
+
 USAGE_STATS_DISABLED_MESSAGE = "Usage stats collection is disabled."
 
-USAGE_STATS_ENABLED_BY_DEFAULT_MESSAGE = (
+USAGE_STATS_ENABLED_BY_DEFAULT_FOR_CLI_MESSAGE = (
     "Usage stats collection is enabled by default without user confirmation "
     "because this stdin is detected to be non-interactively. "
```
Comment (ericl) [inline python/ray/_private/usage/usage_constants.py:28]:
```suggestion
    "because this stdin is detected to be non-interactive. "
```

File: python/ray/_private/usage/usage_constants.py [hunk 3]
```diff
@@ -8,16 +8,22 @@
...
+USAGE_STATS_ENABLED_FOR_CLI_MESSAGE = (
     "Usage stats collection is enabled. To disable this, add `--disable-usage-stats` "
     "to the command that starts the cluster, or run the following command:"
     " `ray disable-usage-stats` before starting the cluster. "
     "See https://docs.ray.io/en/master/cluster/usage-stats.html for more details."
 )
 
+USAGE_STATS_ENABLED_FOR_RAY_INIT_MESSAGE = (
+    "Usage stats collection is enabled. To disable this, run the following command:"
+    " `ray disable-usage-stats` before `ray.init()`. "
```
Comment (ericl) [inline python/ray/_private/usage/usage_constants.py:20]: Users might not have ray.init() in the script, since we do implicit init. For clarity, reword this to

```
Usage stats collection is enabled. To disable, run `ray disable-usage-stats` prior to starting Ray.
```

Here and below.

File: python/ray/_private/usage/usage_lib.py (hunk 1 above)
Comment (jjyao) [reply python/ray/_private/usage/usage_lib.py:375]: It's more of cli vs ray.init rather than nightly vs non-nightly. Nightly cli and non-nightly cli are the same.

**Review by ericl — APPROVED** [review]

**Review by DmitriGekhtman — APPROVED** [review]

File: python/ray/tests/test_usage_stats.py [hunk 4]
```diff
@@ -320,30 +329,103 @@ def test_usage_stats_prompt(
...
+        ray.init()
+        ray.shutdown()
+        captured = capsys.readouterr()
+        assert usage_constants.USAGE_STATS_ENABLED_FOR_RAY_INIT_MESSAGE in captured.out
+        ray.__version__ = saved_ray_version
+        ray.__commit__ = saved_ray_commit
+
+
+def test_is_nightly_wheel():
+    saved_ray_version = ray.__version__
```
Comment (rkooo567) [inline python/ray/tests/test_usage_stats.py:412]: do this in the fixture? (otherwise later tests can be screwed up if this test fails in the middle)

**Review by rkooo567 — APPROVED** [review]

**Review by wuisawesome — APPROVED** [review]
